### ISSUE TITLE
Authenticate to the proxy gateway using a cookie?

### ISSUE THREAD

**Issue by wvezey** [AzureAD/microsoft-authentication-library-for-objc#1113]
TL;DR
Is it possible to authenticate to the proxy gateway using the AzureAppProxyAccessCookie, rather than the Authorization header?

Long Version
Our apps, web and iOS, run within our corporate firewall. Currently, users outside the firewall have to be on the corporate VPN. We are building a POC using this repo as a guide. We are thus far able to successfully authenticate to Azure AD and get the accessToken.

The sample code sends a request to the service layer, authenticating by using the Bearer token in the Authorization header. Our challenge is that we are already using Authorization for our internal service layer authentication, and we understand that the name of the header is not configurable.

We have proven out the end-to-end flow with a web app, and found that authentication through the Azure App Proxy gateway does not use the Authorization header, but rather a cookie named AzureAppProxyAccessCookie is set in the request. Is it possible to use the same approach in an iOS app? If so, is there an MSAL wrapper for managing the cookie?

Any guidance is much appreciated.

**Comment by oldalton** [comment]
Thanks for the question. We don't support issuing a cookie. @brandwe, could you help connecting with the right owners for this use case?

**Comment by wvezey** [comment]
Thanks @oldalton. Headers it is.

**Comment by akvpappan** [comment]
@wvezey how did you manage to resolve the issue? I'm having similar problem. My internal application also expects authentication using the "Authorization: Bearer <>" header.

**Comment by wvezey** [comment]
@akvpappan Our solution involved renaming the header that we use for authenticating requests between our clients and service layer. Thus the header named Authentication is now used exclusively for authenticating to Azure App Proxy. It was a structural change in our app architecture, and thus involved coordination among a couple of teams. Not trivial, but it worked.
